### ISSUE TITLE
Unrelated dialogs should not be dismissed

### ISSUE THREAD

**Issue by stuartmscott** [fyne-io/fyne#2173]
### Describe the bug:
If two unrelated background tasks showing a progress bar dialog and the first task finishes first, both dialogs are dismissed

### Example code:
```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/dialog"
	"fyne.io/fyne/v2/widget"
	"math/rand"
	"time"
)

func main() {
	a := app.New()
	w := a.NewWindow("Overlay Stack")

	go func() {
		time.Sleep(time.Second)
		d := dialog.NewProgressInfinite("Job 1", "Working...", w)
		d.Show()
		time.Sleep(time.Duration(rand.Intn(10)) * time.Second)
		d.Hide()
	}()

	go func() {
		time.Sleep(time.Second)
		d := dialog.NewProgressInfinite("Job 2", "Working...", w)
		d.Show()
		time.Sleep(time.Duration(rand.Intn(10)) * time.Second)
		d.Hide()
	}()

	w.SetContent(widget.NewLabel("Overlay Stack"))
	w.Resize(fyne.NewSize(600, 600))
	w.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** MacOS
 - **Version:** Big Sur
 - **Go version:** go1.15 darwin/amd64
 - **Fyne version:** 72137a5

**Comment by stuartmscott** [comment]
Cross linking to Slack: https://gophers.slack.com/archives/CK5U4BU87/p1618422232128600

**Comment by stuartmscott** [comment]
In the above, if Job 1 finishes quickly and Job 2 is a long running task the user is expected to see the progress dialog for Job 2 until it completes. However when Job 1 completes and its dialog is dismissed, Job 2's dialog is also removed.

**Comment by andydotxyz** [comment]
Hmm, this is a tricky one. I don't think we've discussed supporting multiple dialogs open at the same time.

**Comment by stuartmscott** [comment]
I agree that while the toolkit supports multiple dialogs open at the same time, the user should only be able to interact with one. But why should closing one dialog also close another?

**Comment by andydotxyz** [comment]
Dismissing child elements is important so that things like menus, select dropdowns and other multi-layer elements dismiss cleanly. The overlay stack is like a single branch of the object tree, the item prior in the list is considered a parent.
We certainly could revisit this, but I think it is more complex than simply changing the current behaviour to fix this issue, it would essentially change the overlays contract so we'd need to do a more thorough exploration of the ramifications.

**Comment by beoran** [comment]
The solution in order to keep a tree-like stucture would be to add to Fyne an (invisible?) parent for both dialogs. Something like a "Desktop" or "MDI" widget or so.

**Comment by andydotxyz** [comment]
I don't really understand this comment sorry @beoran, our dialogs have OS level windows as their parent - we don't support "floating window dialogs" like some OS have.
